src/info/global_stats.h

```diff
@@ -44,6 +44,13 @@ typedef struct {
   size_t numVectorFieldsHNSW;
   size_t numVectorFieldsSvsVamana;
   size_t numVectorFieldsSvsVamanaCompressed;
+  // Total number of documents indexed by each field type
+  size_t textTotalDocsIndexed;
+  size_t tagTotalDocsIndexed;
+  size_t numericTotalDocsIndexed;
+  size_t geoTotalDocsIndexed;
+  size_t geometryTotalDocsIndexed;
+  size_t vectorTotalDocsIndexed;
 } FieldsGlobalStats;
```

Comment:
What is the thread-safty guarentee and assumption to update these metrics?
Can they be updated from the bg by the background indexer and read from the main thread?